packages/components/src/higher-order/with-focus-outside/index.js

```diff
@@ -1,142 +1,37 @@
-/**
- * External dependencies
- */
-import { includes } from 'lodash';
-
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useCallback, useState } from '@wordpress/element';
 import { createHigherOrderComponent } from '@wordpress/compose';
 
 /**
- * Input types which are classified as button types, for use in considering
- * whether element is a (focus-normalized) button.
- *
- * @type {string[]}
- */
-const INPUT_BUTTON_TYPES = [ 'button', 'submit' ];
-
-/**
- * Returns true if the given element is a button element subject to focus
- * normalization, or false otherwise.
- *
- * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus
- *
- * @param {Element} element Element to test.
- *
- * @return {boolean} Whether element is a button.
+ * Internal dependencies
  */
-function isFocusNormalizedButton( element ) {
-	switch ( element.nodeName ) {
-		case 'A':
-		case 'BUTTON':
-			return true;
-
-		case 'INPUT':
-			return includes( INPUT_BUTTON_TYPES, element.type );
+import { useFocusOutside } from '../../utils/hooks';
+
+export default createHigherOrderComponent(
+	( WrappedComponent ) => ( props ) => {
+		const [ handleFocusOutside, setHandleFocusOutside ] = useState();
+		const bindFocusOutsideHandler = useCallback(
+			( node ) =>
+				setHandleFocusOutside( () =>
+					node?.handleFocusOutside
+						? node.handleFocusOutside.bind(
+								node,
+								node.handleFocusOutside
+						  )
+						: undefined
+				),
+			[ setHandleFocusOutside ]
```

Comment:
`setHandleFocusOutside` won't change.

```suggestion
			[]
```